Contributing.md

```diff
@@ -1,30 +1,471 @@
+
+Almost all new code enters the codebase via the `development` branch. Hotfixes and urgent updates can be made on 
+other branches from time to time. These changes will then be cherry-picked into the other branches to keep 
+the code base consistent.
+
+## Testnets
+
+There will also be three testnets in addition to mainnet:
+
+* `mainnet` - the "real-money" network.
+* `stagenet` - a testnet version of mainnet. Nodes on stagenet should all be running code built from the `mainnet`
+  branch to replicate mainnet as close as possible. The base node version on `stagenet` will always match
+  the latest `mainnet` release. Network resets (i.e. a new genesis block) should be very rare.
+* `nextnet` - nodes on nextnet should be running code compiled from the `nextnet` branch. This is the last chance to
+  battle-test new features before they hit mainnet. Network resets should be somewhat rare.
+* `dibbler` - The dibbler network is the most experimental of the testnets. Things can break here. We hope they 
+  won't, obviously, but the dibbler network is where we should be catching any and all major issues _before_ 
+  the feature heads to nextnet. Nodes should be running code built from the `dibbler` branch. Network resets could 
+  occur fairly frequently, perhaps several times a year.
+
+## Feature flags
+
+When developing a new feature in Tari, it will initially be placed behind a feature flag -- similar to how Rust 
+nightly features are gated. This offers the comfort and flexibility that should a feature not be ready in the first 
+6-8 months of development (i.e. by the time it first hits the `mainnet` branch), it will not be active on the 
+mainnet, since the feature will not be active.
+
+The name of the feature flag is assigned in the tracking issue 
+of the RFC that describes the feature. Once the feature is complete and ready for testing, it can be enabled in the 
+next release of `dibbler`. A feature can be tested on the `dibbler` testnet for as long as necessary. Even though the 
+code for this feature will be included in the following `nextnet` release, the feature will not be active. 
+Once the community feels that the feature is ready for mainnet, it will be activated on `nextnet`. This gives us 
+another 2 months of testing in an environment that is reasonably close to what might be expected on mainnet. 
+
+As the next release approaches, the community will decide whether to "stabilise" the feature, i.e. remove the 
+feature gate. The feature will then be live on mainnet with the next release. If the feature is not 
+stabilised, it remains in testing for another cycle.
+
+Notice that stabilising involves removing the feature gate entirely so that the code is included by default. Put 
+another way, the code running on `mainnet` is always the code that compiles with the default set of features.  
+
+## Versioning
+
+The release period is **TWO MONTHS**.
+
+At every release period, 
+* the HEAD of `development` becomes `dibbler`.
```

Comment:
Is there a set point in time where features on dibbler are enabled/disabled, or can that happen at any time in between release periods?

Reply:
They can be activated any time.